config/config.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"flag"
+	"github.com/caarlos0/env"
+	"github.com/northmule/gophermart/internal/app/services/logger"
+	"os"
+	"strings"
+)
+
+const DataBaseConnectionTimeOut = 10000
+const serverURLDefault = ":8081"
+const databaseURIDefault = "postgres://postgres:123@localhost:5456/gofermart?sslmode=disable"
+const accrualURLDefault = "http://localhost:8091"
+
+type GophermartConfig struct {
+	// Адрес сервера и порт
+	ServerURL string `env:"RUN_ADDRESS"`
+	// Внешняя система расчёта бонусов
+	AccrualURL string `env:"ACCRUAL_SYSTEM_ADDRESS"`
+	// Строка подключения к БД
+	DatabaseURI string `env:"DATABASE_URI"`
+}
+
+func NewGophermartConfig() (*GophermartConfig, error) {
+	instance := &GophermartConfig{}
+	err := instance.env()
+	if err != nil {
+		return nil, err
+	}
+
+	return instance, instance.flag()
+}
+
+func (c *GophermartConfig) env() error {
+	err := env.Parse(c)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *GophermartConfig) flag() error {
+	cf := flag.FlagSet{}
+	serverURL := cf.String("a", serverURLDefault, "адрес и порт запуска сервиса")
+	databaseURI := cf.String("d", databaseURIDefault, "адрес подключения к базе данных")
+	accrualURL := cf.String("r", accrualURLDefault, "адрес системы расчёта начислений")
+
+	err := cf.Parse(os.Args[1:])
+	if err != nil {
+		logger.LogSugar.Errorf("флаги конфигурации не разобраны: %s", err)
+		return err
```

Comment:
Не обязательно логировать прямо в момент возникновения ошибки. Бывает удобнее просто обернуть ошибку, прокинуть вверх по стеку и залогировать там. 
```suggestion
		return fmt.Errorf("флаги конфигурации не разобраны: %w", err)
```